src/lib.rs

```diff
@@ -369,20 +337,17 @@ impl GrowableBloom {
     /// assert!(existed_before == true);
     /// ```
     pub fn check_and_set<T: Hash>(&mut self, item: T) -> bool {
-        let prev = self.contains(&item);
-        if !prev {
-            self.insert(item);
-        }
-        prev
+        !self.insert(item)
     }
 
     /// Grow the GrowableBloom
     fn grow(&mut self) {
-        self.curr_num_slice += 1;
-        self.slice_size *= GrowableBloom::GROWTH_FACTOR;
-        self.curr_seed ^= hash_u64(self.curr_seed);
-        let new_bloom = Bloom::new(self.curr_num_slice, self.slice_size, self.curr_seed);
-        self.blooms.push(new_bloom)
+        let error_ratio =
+            self.desired_error_prob * Self::TIGHTENING_RATIO.powi(self.blooms.len() as _);
```

Comment:
Would you mind leaving a comment explaining the idea behind this? You have a comment on one of the tests that you could re-use. It's been a long time since I read the paper, and it'll probably confuse newcomers as well